Content.Client/DeltaV/CartridgeLoader/Cartridges/StockTradingUiFragment.xaml.cs

```diff
@@ -0,0 +1,269 @@
+                MinWidth = 140,
+            };
+
+            _buyButton = new Button
+            {
+                Text = "Buy",
+                MinWidth = 65,
+                Margin = new Thickness(3, 0, 3, 0),
+            };
+
+            _sellButton = new Button
+            {
+                Text = "Sell",
+                MinWidth = 65,
+            };
+
+            buttonContainer.AddChild(_buyButton);
+            buttonContainer.AddChild(_sellButton);
+
+            bottomRow.AddChild(_sharesLabel);
+            bottomRow.AddChild(_amountEdit);
+            bottomRow.AddChild(buttonContainer);
+
+            // Add the bottom row last
+            mainContent.AddChild(bottomRow);
+
+            companyPanel.AddChild(mainContent);
+            Container.AddChild(companyPanel);
+
+            // Add horizontal separator after the panel
+            var separator = new HSeparator
+            {
+                Margin = new Thickness(5, 3, 5, 5),
+            };
+            Container.AddChild(separator);
+
+            // Button click events
+            _buyButton.OnPressed += _ =>
+            {
+                if (float.TryParse(_amountEdit.Text, out var amount) && amount > 0)
+                    onBuyPressed?.Invoke(companyName1, amount);
+            };
+
+            _sellButton.OnPressed += _ =>
+            {
+                if (float.TryParse(_amountEdit.Text, out var amount) && amount > 0)
+                    onSellPressed?.Invoke(companyName1, amount);
+            };
+
+            // There has to be a better way of doing this
+            _amountEdit.OnTextChanged += args =>
+            {
+                var newText = string.Concat(args.Text.Where(char.IsDigit));
+                if (newText != args.Text)
+                    _amountEdit.Text = newText;
+            };
+        }
+
+        public void Update(StockCompanyStruct company, int ownedStocks)
+        {
+            _nameLabel.Text = company.DisplayName;
+            _priceLabel.Text = $"${company.CurrentPrice:F2}";
+            _sharesLabel.Text = $"Owned: {ownedStocks}";
+
+            var priceChange = 0f;
+            if (company.PriceHistory.Count > 0)
+            {
+                var previousPrice = company.PriceHistory[^1];
+                priceChange = (company.CurrentPrice - previousPrice) / previousPrice * 100;
+            }
+
+            _changeLabel.Text = $"{(priceChange >= 0 ? "+" : "")}{priceChange:F2}%";
+
+            // Update color based on price change
+            _changeLabel.Modulate = priceChange switch
+            {
+                > 0 => PositiveColor,
+                < 0 => NegativeColor,
+                _ => NeutralColor,
+            };
+
+            // Update the price history table
+            _priceHistory.Update(company.PriceHistory);
+
+            // Disable sell button if no shares owned
+            _sellButton.Disabled = ownedStocks <= 0;
+        }
+    }
+}
```

Comment:
This file has a lot of hard coded text that will be an issue in Russian etc.

Reply:
i really didn't want to do localization for all of this raaaaaaghh